language-server/src/util/util.js

```diff
@@ -108,3 +113,40 @@ export const readDirRecursive = async function* (path, filter, cwd) {
     }
   }
 };
+
+/** @type (textDocument: TextDocument, textEdit: TextEdit, settings: IndentationSettings) => TextEdit */
+export const withFormatting = (textDocument, textEdit, settings) => {
+  const indentation = settings.detectIndentation ? detectIndent(textDocument.getText()) : {
+    amount: settings.tabSize,
+    type: settings.insertSpaces ? "space" : "tab"
+  };
+
+  const formattingOptions = {
+    insertSpaces: indentation.type === "space",
+    tabSize: indentation.amount,
+    keepLines: true,
+    eol: settings.endOfLine
```

Comment:
I noticed that the `files.eol` setting can be "auto", which is not a valid value here. Node has `os.EOL` that we can use if `files.eol` is "auto".